mistakedetection/src/ca/mcgill/sel/mistakedetection/MistakeDetection.java

```diff
@@ -2643,8 +2655,9 @@ public static Optional<Mistake> checkMistakeUppercaseAttribName(Attribute studen
   }
 
   public static Optional<Mistake> checkMistakeClassSpelling(Classifier studentClass, Classifier instructorClass) {
-    if (spellingMistakeCheck(studentClass.getName(), instructorClass.getName()) && !isPlural(studentClass.getName())
-        && !isLowerName(studentClass.getName())) {
+    int lDistance = levenshteinDistance(studentClass.getName(), instructorClass.getName());
+    if (isSpelledIncorrectly(lDistance, studentClass.getName().length(), instructorClass.getName().length()) && !isPlural(studentClass.getName())
+        && !isLowerName(studentClass.getName()) && !isSoftwareEngineeringTerm(studentClass.getName())) {
```

Comment:
There are 5 calls to `studentClass.getName()` in the above lines so I would extract it to a variable to improve readability